egs/wsj/s5/steps/cleanup/segment_long_utterances.sh

```diff
@@ -50,15 +50,17 @@ cmd=run.pl
 
 . utils/parse_options.sh
 
-if [ $# -ne 6 ]; then
+if [ $# -ne 5 ] && [ $# -ne 7 ]; then
     cat <<EOF
-Usage: $0 [options] <model-dir> <lang> <data-in> <text-in> <segmented-data-out> <work-dir>
+Usage: $0 [options] <model-dir> <lang> <data-in> [<text-in> <utt2text>] <segmented-data-out> <work-dir>
  e.g.: $0 exp/wsj_tri2b data/lang_nosp data/train_long data/train_long/text data/train_reseg exp/segment_wsj_long_utts_train
 This script performs segmentation of the data in <data-in> and 
 transcript <text-in>, writing the segmented data (with a segments file) to
 <segmented-data-out> along with the corresponding aligned transcription.  
-Note: <text-in> must be indexed by the utterance-ids of the utterances in
-<data-in>
+Note: If <utt2text> is not provided, the text in <data-in> is used as the 
+original transcript.
+If <utt2text>, if provided, is a mapping from the utterance in <data-in> to the 
+<original-transript> in <text-in>.
```

Comment:
you use the token original-transcript without defining what it is.  I assume you refer to the key that is
the first field of text-in.  Maybe "transcript-key" would be clearer, as it clarifies that it is a key to a
transcript and not an actual transcript.